### PR TITLE
Release script prompts for NPM 2FA code

### PR DESCRIPTION
**Pull request by bvaughn**
I haven't _really_ tested this, but in theory, it prompts you for a 2FA code prior to publishing. If you provide a code, it passes it via the `--opt` flag to `npm publish` which is how the docs specify you should pass it.

Publishing is quick, and hopefully all of the packages can be published with the same token. But it's possible that it will expire in the middle, which would not be ideal. In that case, you would need to re-run the publish command with modifications to skip the already published packages.

I'm opening this PR for discussion purposes. I'll give it a little more thought before marking it ready for review.

### CODE REVIEW

File: scripts/release/publish-commands/publish-to-npm.js [hunk 1]
```diff
@@ -19,10 +19,17 @@ const push = async ({cwd, dry, packages, version, tag}) => {
     throw new Error('The tag `latest` can only be used for stable versions.');
   }
 
+  // Pass two factor auth code if provided:
+  // https://docs.npmjs.com/getting-started/using-two-factor-authentication
+  const twoFactorAuth = opt != null ? `--opt ${opt}` : '';
```
Comment (sophiebits) [inline scripts/release/publish-commands/publish-to-npm.js]: otp?

Comment (bvaughn) [reply scripts/release/publish-commands/publish-to-npm.js]: Gah! I read it, multiple times, as "opt" 😦 but you're right.

**Comment by bvaughn** [comment]
> Publishing is quick, and hopefully all of the packages can be published with the same token. But it's possible that it will expire in the middle, which would not be ideal. In that case, you would need to re-run the publish command with modifications to skip the already published packages.

I think this is okay for now. It's already the case if you e.g. don't have access to a package and it fails midway through. So let's follow up if we want to clean up this in general.

**Comment by sophiebits** [comment]
I don't expect it to be a large problem in practice. In practice, tokens are good for a minute or two in order to support clock drift.

**Comment by bvaughn** [comment]
Would you...?

**Comment by bvaughn** [comment]
Thanks!

**Pull request merged**